### DISCUSSION TITLE
[JOBS] How to complete a task from an outside command

### DISCUSSION THREAD

**Discussion by attisbarros** [roadrunner-server/roadrunner#1512] [Q&A] [answered]
It's possible I'm misusing a queue system as I haven't yet fully understood it's full capabilities or usefulness. With that said, I understand I can create a queue to proccess events such as sending email notifications, but none of the examples I've searched uses it to proccess long running tasks.

Basically I need to request financial data from a broker every minute for a specific symbol, run that data through my strategy algorithm, (optionally publish it to centrifuge, whenever I decide to watch the graph on the front-end), until I decide to terminate the [session](https://i.imgur.com/clVOhQ5.png).

Up to this point, I've writen the code to create a session, and send that session payload to the worker - now as far as I can see there's no way to terminate/complete or even fail a specific job after it's been sent to the queue, I figured the [task Id](https://roadrunner.dev/docs/plugins-jobs/2.x/en#received-task-id) could be used to fetch it from the queue but there's nothing in the documentation in that regard.

If I have multiple sessions (running in the background) the only possible way is to [terminate/pause/resume](https://github.com/roadrunner-server/roadrunner/issues/1162) the entire queue but not any specific job within that queue.

Was hoping someone could lead me in the right direction.

**Comment by rustatian** [comment, marked as answer]
Hey @att1sb 👋🏻 
I guess, that it would be better to use the `service` plugin: https://roadrunner.dev/docs/plugins-service/2.x/en
Or, much better would be to use the temporal php-sdk with the RR: https://github.com/temporalio/sdk-php

**Reply by rustatian** [reply]
But, unfortunately, this is not possible to stop the job when RR has already consumed it from the driver (like rabbitmq, beanstalk, etc). But this is definitely possible in the temporal + you would have the async runtime 😃

**Reply by attisbarros** [reply]
Perfect Valery, appreciate it!
